### PR TITLE
Aea 410 strict ts

### PR DESCRIPTION
**Pull request by dan-jones-iw**
3 separate PRs are in one here because i keep pushing changes before they are merged. sorry about the amount of changes, happy to walk through it. if anyone has suggestions on how to break up the code into separate PRs im happy to do that too. 

in summary - changed the fields in our model of MedicationRequest to be consistent with the MiM or the IG in terms of if they should be present or not. as well as that the MedicationRequest model is an interface, and so ive had to change various parts of the codebase to reflect that. happy to go into more detail in a call.

### CODE REVIEW

File: coordinator/tests/services/translation/common/getResourcesOfType.spec.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  test("getPatient", () => {
+    const patient = getPatient(bundle)
+
+    expect(patient.resourceType).toBe("Patient")
+  })
+
+  test("getProvenance", () => {
+    const provenances = getProvenances(bundle)
+
+    provenances.map((provenance) => expect(provenance.resourceType).toBe("Pro"))
```
Comment (booshibooshi) [inline coordinator/tests/services/translation/common/getResourcesOfType.spec.ts:26]: ResourceType should be "Provenance", and the bundle here should be fhirMessageSigned from TestResources

Comment (booshibooshi) [reply coordinator/tests/services/translation/common/getResourcesOfType.spec.ts:26]: Before the refactor we were asserting the length of the array returned, maybe we should add this check back in to ensure we're not getting any false positives for tests as in this case

**Review by booshibooshi — CHANGES_REQUESTED** [review]

File: coordinator/tests/services/translation/common/getResourcesOfType.spec.ts (hunk 1 above)
Comment (dan-jones-iw) [reply coordinator/tests/services/translation/common/getResourcesOfType.spec.ts:26]: ahh yeah this is something i left and never came back to, thanks for spotting!

Comment (dan-jones-iw) [reply coordinator/tests/services/translation/common/getResourcesOfType.spec.ts:26]: maybe we should have the function throw an error if it doesnt bring back a resource? can chat about it

Comment (booshibooshi) [reply coordinator/tests/services/translation/common/getResourcesOfType.spec.ts:26]: discussed and agreed that moving the expectations of number of entries returned for a resource type made sense to be asserted during validation

**Review by booshibooshi — APPROVED** [review]

**Pull request merged**